File: 2a.py
data = [tuple(line.split(" ")) for line in open("input/2.txt").read().split("\n")[:-1]]

horiz = 0
depth = 0

for (dir, units) in data:
    if(dir[0] == "f"):
        horiz += int(units)
    if(dir[0] == "d"):
        depth += int(units)
    if(dir[0] == "u"):
        depth -= int(units)

print(horiz * depth)

File: 2a.kt
import java.io.File

enum class DirType {
    Forward,
    Down,
    Up;

    companion object {
        fun from(str: String): DirType {
            return when (str.get(0)) {
                'f' ->  DirType.Forward
                'd' ->  DirType.Down
                'u' ->  DirType.Up
                else -> DirType.Forward
            }
        }
    }
}

fun main() {
    var commands = ArrayList<Pair<DirType, Int>>()
    var horiz = 0
    var depth = 0

    File("./input/2.txt").forEachLine { line: String ->
        commands.add(Pair(DirType.from(line), line.substringAfter(' ').trim().toInt()))
    }

    for((type, units) in commands) {
        when (type) {
            DirType.Forward -> horiz += units
            DirType.Down -> depth += units
            DirType.Up -> depth -= units
        }
    }

    println(horiz * depth)
}